MenuTitle: new component & MenuSeparator: remove title and set correct style

REJack opened the pull request:
I've added a new component MenuTitle & removed the title from MenuSeparator.

I've removed your default uppercase class to comply with daisyUI default style and it's hard to remove it if you don't want it (like me).

Example Images from maryUI-Docs - Left (modified) / Right (current)
![image](https://github.com/robsontenorio/mary/assets/1554582/9f7223cd-fd74-4ed7-b28e-2f6dc439b475) ![image](https://github.com/robsontenorio/mary/assets/1554582/ec0a3a21-ecb6-44a7-b105-414a9596cb8a)
![image](https://github.com/robsontenorio/mary/assets/1554582/6659007b-e251-4ce8-a863-a82a600841da) ![image](https://github.com/robsontenorio/mary/assets/1554582/34f5167c-a5ca-427a-b498-c0c8426a3013)

In my use-case it looks like that

Dropdown with a simple MenuTitle and a single MenuSeparator
![image](https://github.com/robsontenorio/mary/assets/1554582/1ea0a5fb-279e-433e-b897-19fe96521fbf)


P.S.: mary-ui.com change/PR is coming too ;)

robsontenorio commented:
@REJack 

Why not modify the current `x-menu-separator` to have better look and feel? As I can see it would be just the "divider" style.

It currently has all necessary attributes (optional), so we would not need a new component.

REJack commented:
I've updated my PR, with that change my docs PR is not needed too, now I've updated the style + we keep the old title icon feature but with usage of my new MenuTitle. 

What you think about that?

I would still prefer a MenuTitle component to only have a title without separator/divider 😄.

robsontenorio commented:
Agreed. Reverted to the first commit. Thanks!
